src/librustc_typeck/astconv.rs

```diff
@@ -1265,6 +1266,7 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx> + 'o {
                                      ref_id: ast::NodeId,
                                      span: Span,
                                      ty: Ty<'tcx>,
+                                     ty_hir: &hir::Ty,
```

Comment:
Unnecessary, see https://github.com/rust-lang/rust/pull/56225/commits/a15b1c92b38475f9d64e6418bfb604ae11d17bc9?w=1#r239269084.

Reply:
How is it unnecessary? It's used for dignostics purposes.

Reply:
I mean, *will* be unused/unnecessary, when the feature gating issue is addressed.